### PR TITLE
Go module build improvements

### PR DESCRIPTION
**Pull request by pgaskin**
- Add module info to version
- Add warning if built without Go modules
- Update source installation instructions for Go modules

see #890

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -15,6 +16,9 @@ import (
 
 func main() {
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
+	if info, ok := debug.ReadBuildInfo(); !ok && info == nil {
+		fmt.Fprint(os.Stderr, "Warning: dnscontrol was built without Go modules. See https://github.com/StackExchange/dnscontrol#from-source for more information on how to build dnscontrol correctly.\n\n")
```
Comment (pgaskin) [inline main.go:20]: @tlimoncelli, what do you think of this message?

File: main.go [hunk 2]
```diff
@@ -36,6 +40,9 @@ func versionString() string {
 	} else {
 		version = fmt.Sprintf("%s-dev", Version) // no SHA. '0.x.y-dev' indicates it is run from source without build script.
 	}
+	if info, ok := debug.ReadBuildInfo(); !ok && info == nil {
+		version += " (non-modules)"
```
Comment (pgaskin) [inline main.go:44]: Note that it's also possible to list all module versions if that's something you're interested in.

File: main.go (hunk 1 above)
Comment (tlimoncelli) [reply main.go:20]: :+1:

File: main.go (hunk 2 above)
Comment (tlimoncelli) [reply main.go:44]: We can add that later. I think this is good for now.

**Review by tlimoncelli — APPROVED** [review]

**Pull request merged**